src/provisioner/provisioner.py

```diff
@@ -221,12 +252,71 @@ def cluster_upgrade(cortx_conf_url: str):
         Paramaters:
         [IN] CORTX Config URL
         """
-        cortx_conf = ConfigStore(cortx_conf_url)
-
-        node_id, node_name = CortxProvisioner._get_node_info(cortx_conf)
+        config_store = ConfigStore(config_store_url)
+        phase_name = const.PROVISIONING_STAGES.UPGRADE.value
+        node_id, node_name = CortxProvisioner._get_node_info(config_store)
+        is_valid = CortxProvisioner._validate_provisioning(
+            config_store, node_id, phase_name)
+        if is_valid is False:
+            Log.info('Validation check failed, Aborting cluster upgrade.')
+            return 0
         Log.info(f"Starting cluster upgrade on {node_id}:{node_name}")
         # TODO: validations
 
         mp_interfaces = ['upgrade']
-        CortxProvisioner._provision_components(cortx_conf, mp_interfaces)
-        Log.info(f"Finished cluster upgrade on {node_id}:{node_name}")
\ No newline at end of file
+        CortxProvisioner._provision_components(config_store, mp_interfaces, phase_name)
+        Log.info(f"Finished cluster upgrade on {node_id}:{node_name}")
+
+    @staticmethod
+    def _update_provisioning_info(config_store, node_id, phase,
```

Comment:
Please rename to _update_provisioning_status

Reply:
@nitin-seagate Suggested to change this to _update_provisioning_info as we are updating phase and status both information.
And not only the status key.

Reply:
Still at the macro level its status so we should use _update_provisioning_status (here status is not the status as key)

Reply:
DOne, Renamed to _update_provisioning_status.